src/components/InputV2/Button.tsx

```diff
@@ -0,0 +1,72 @@
+import PropTypes from 'prop-types'
+import { BaseInput, BaseInputProps, BaseInputPropTypes } from './Base'
+
+/**
+ * Button
+ */
+
+const buttonClassNames = ['disabled', 'large', 'white', 'gray', 'red', 'green']
+
+interface ButtonClassNameProps {
+  disabled?: boolean
+  large?: boolean
+  white?: boolean
+  gray?: boolean
+  red?: boolean
+  green?: boolean
+}
+
+const ButtonClassNamePropTypes = {
+  disabled: PropTypes.bool,
+  large: PropTypes.bool,
+  white: PropTypes.bool,
+  gray: PropTypes.bool,
+  red: PropTypes.bool,
+  green: PropTypes.bool
+}
+
+const getButtonClassName = (props: ButtonClassNameProps): string => {
+  return Object.keys(props)
+    .filter(name => props[name] && buttonClassNames.includes(name))
+    .join(' ')
+}
```

Comment:
@Schwartz10 just FYI,

This part converts the passed props to HTML classnames, which will allow us to refactor `ButtonV2` completely so it uses the same styling from `base-css`. Then it doesn't need to be a styled component anymore.

However, we first need to phase out all the old `Button` components and remove any styled system declarations from the `ButtonV2` components. Then we can apply the button styling to `button` elements and `a.button` elements (for `ButtonV2Link`). Currently this is limited to `input[type="button"]` so we don't mess with existing components.

```
<button className='red'>Cancel</button>
```